pkg/controller/job/job_controller.go

```diff
@@ -143,6 +144,13 @@ type syncJobCtx struct {
 	ready                           int32
 }
 
+type orphanPodKey struct {
+	// Either "key" or "selector"
+	kind string
+	// Either "namespace/name" (for kind key) or "namespace:selector" (for kind selector)
```

Comment:
if there is a namespace for both, I suggest storing it in its own field, rather than concatenating it.

Reply:
We don't technically need the namespace for key kind: name in orphan reconciler operations. However, during the PR discussion, we organically agreed to include it for logging purposes since we're already supplying it for key kind: selector. Given that, it made sense to also include it for key kind: name, so moving the namespace to its own field seems right to me now.